### PR TITLE
Categories page

### LINKED ISSUE #36: Categories Page Component
Create this page, and make it dynamically fetch/display all the categories. Can exclude the sort order for now, and no need to make the edit button work for now, but make the delete button work as well. Note this page should be created as a component, as it also exists in the global search view by category section, so it should be able to display all the categories with type "Principles" or all categoreis with type "Emergencies". It should also be able to display all the emergency pages, not by category. Add backend routes if needed for those. Check figma for more specifics. 

Principles view:
image

Emergency categories + all view: 
image

### PR DESCRIPTION
**Pull request by EdwardLinS**
## Tracking Info

Resolves #36 

## Changes

- Added the emergency and general principle category pages to display all categories under each
- Added the all pages page to show all pages and which category they belong to
- Allow deletion of categories and pages
- Create a toast to confirm the deletion of categories

## Testing

- Tested fetching each category
- Tested fetching each page
- Testing deleting pages and categories
- Changed the size of the window to test responsiveness

## Confirmation of Change

image
image
image
image

### CODE REVIEW

**Review by eshaan-s18 — APPROVED** [review]
This is awesome! Great job! The dynamic fetching and displaying of all the categories works great for me. The pages look great and follow the Figma well! Also the resizing of the window feature looks great too! The delete button on the Emergencies and General Principles pages work seamlessly; the only thing I noticed was that the delete button on the All Pages page doesn't seem to be functioning for me. Not sure if this was intended or not. I'll approve this PR though because everything else is working great and making the delete button work shouldn't be too challenging as you have it already implemented on the other pages. If this was intentional though, then everything looks good to me! Great job team!

File: admin-portal-frontend/src/app/components/categoryRoutes.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  _id: string;
+  title: string;
+  items: [];
+  type: string;
+};
+
+// Gets and returns all categories from API
+export const getAllCategories = async () => {
+  try {
+    const url = "http://localhost:3001/api/categories";
```
Comment (Anthonyp0329) [inline admin-portal-frontend/src/app/components/categoryRoutes.tsx]: have the url domain be stored in a .env, just for future production deployment and such

**Review by Anthonyp0329 — APPROVED** [review]
Awesome work guys!! Looks good to go!! For the purposes of merging into main, make sure to update the branch with latest commits, and you can get rid of the changes to the main page.tsx for now. @pratyush1718 will take care of having the navbar route to the appropriate page. Thanks!!

**Review by r800360 — CHANGES_REQUESTED** [review]
Awesome PR Edward and Lillian! Changes look great to me and using the recent environmental variable change, I'm able to connect in the hosted backend: https://sideline-sidekick-app.web.app/api/categories as well to fetch the data for the three pages: all-page, emergencies, general-principles. 

As Eshaan mentioned, could you investigate the issue with the delete button at the all-page (http://localhost:3000/all-page/)? Other than that, everything else looks good. Thanks in advance and sorry for the late review.

**Review by r800360 — APPROVED** [review]
Edward and Lillian, nice job fixing the delete button on the all-page as well as all of those nasty linter errors! This code looks good to merge to me.

**Review by eshaan-s18 — APPROVED** [review]
Looks good to me! Great job!

**Review by Anthonyp0329 — APPROVED** [review]
Looks good and ready to merge! Awesome work!

**Pull request merged**